src/abi.cpp

```diff
@@ -65,6 +66,28 @@ abi_type* get_type(std::map<std::string, abi_type>& abi_types, const std::string
             );
             auto [iter, success] = abi_types.try_emplace(name, name, abi_type::array{element}, &abi_serializer_for< ::abieos::pseudo_array>);
             return &iter->second;
+        } else if (ends_with(name, "]")) {
+            // fixed_array
+            int size = 0;
+            if (auto idx = name.find_last_of('['); idx != std::string::npos) {
+                auto fc_res = std::from_chars(&name[idx + 1], &name[name.size() - 1], size);
+                check (fc_res.ec == std::errc{}, "Unexpected size specification for fixed array type");
+                check(fc_res.ptr == &name[name.size() - 1], "Unexpected size specification for fixed array type");
+                if (size <= 0) {
+                    eosio::check(size != 0, "Zero size fixed arrays not allowed");
+                    eosio::check(size > 0, "Negative size fixed arrays not allowed");
```

Comment:
My understanding is that `std::from_chars` will not allow negative if you use `uint32_t` instead of `int`.